### PR TITLE
[rfr] Calling Node#update with same data doesn't generate multiple logs

### PR DESCRIPTION
**Pull request by sloria**
### Purpose

Sending multiple `PATCH /v2/nodes/<nid>` requests with the same category generated multiple logs, which was inconsistent with the behavior for PATCHing title and description.

### Changes

Modify `Node#update` to only create a log when a field has been changed.

### Ticket

https://openscience.atlassian.net/browse/OSF-4664

### CODE REVIEW

**Comment by brianjgeiger** [comment]
Looks good, works well. :penguin:

**Comment by abought** [comment]
Can you please merge conflicts and ping me when this is ready to review?

File: tests/api_tests/nodes/test_views.py [hunk 1]
```diff
@@ -1086,6 +1086,24 @@ def test_partial_update_private_project_logged_in_non_contributor(self):
...
+
+        res = self.app.patch_json_api(self.private_url, payload, auth=self.user.auth)
+        self.private_project.reload()
+        assert_equal(self.private_project.category, new_category)
+        assert_equal(len(self.private_project.logs), original_n_logs + 1)  # sanity check
+
+        res = self.app.patch_json_api(self.private_url, payload, auth=self.user.auth)
+        self.private_project.reload()
+        assert_equal(self.private_project.category, new_category)
+        self.private_project.reload()
```
Comment (abought) [inline tests/api_tests/nodes/test_views.py]: This `reload` statement may be unnecessary.

File: tests/test_models.py [hunk 2]
```diff
@@ -1941,6 +1941,24 @@ def test_updating_description_twice_with_same_content(self):
...
+        new_category = 'data'
+
+        self.node.update({'category': new_category}, auth=Auth(self.user), save=True)
+        assert_equal(len(self.node.logs), original_n_logs + 1)  # sanity check
+        assert_equal(self.node.category, new_category)
+
+        # Call update with same category
+        self.node.update({'category': new_category}, auth=Auth(self.user), save=True)
+
+        # A new log is not created
```
Comment (abought) [inline tests/test_models.py]: Can you reword this to "only one new log is created" for clarity?

File: tests/api_tests/nodes/test_views.py [hunk 3]
```diff
@@ -1086,6 +1086,24 @@ def test_partial_update_private_project_logged_in_non_contributor(self):
...
         assert_in('detail', res.json['errors'][0])
 
+    def test_multiple_patch_requests_with_same_category_generates_one_log(self):
+        self.private_project.category = 'project'
+        self.private_project.save()
+        new_category = 'data'
+        payload = make_node_payload(self.private_project, attributes={'category': new_category})
+        original_n_logs = len(self.private_project.logs)
+
+        res = self.app.patch_json_api(self.private_url, payload, auth=self.user.auth)
```
Comment (abought) [inline tests/api_tests/nodes/test_views.py]: For clarity, please add a check that the return code after each request is also what you'd expect.

(the lack of `expect_errors` implies the request succeeded, but it's good practice to explicitly ensure that tests work for the reason we expect)

**Comment by abought** [comment]
CR2 pass complete. Just a few minor cleanup items, plus some merge conflicts to address.

**Comment by sloria** [comment]
Changes made. Thanks.
